framework/src/task_manager/task_manager_core.c

```diff
@@ -447,6 +447,67 @@ static void taskmgr_termination_callback(void)
 	mq_unlink(TM_PUBLIC_MQ);
 }
 
+static int taskmgr_check_msg_mask(int msg_mask, int handle)
+{
+	if (msg_mask & TASK_MSG_MASK(handle)) {
+		return OK;
+	}
+	return ERROR;
+}
+
+static int taskmgr_get_handle_by_pid(int pid)
+{
+	int chk_idx;
+	for (chk_idx = 0; chk_idx < CONFIG_TASK_MANAGER_MAX_TASKS; chk_idx++) {
+		if (TASK_PID(chk_idx) == pid) {
+			return chk_idx;
+		}
+	}
+	return ERROR;
+}
+
+static void taskmgr_broadcast(int msg)
+{
+	int chk_idx;
+	int fd;
+	int ret;
+
+	fd = taskmgr_get_drvfd();
+	union sigval msg_broad;
+	msg_broad.sival_int = msg;
+	for (chk_idx = 0; chk_idx < CONFIG_TASK_MANAGER_MAX_TASKS; chk_idx++) {
+		if (TASK_LIST_ADDR(chk_idx) != NULL) {
+			ret = taskmgr_get_task_state(chk_idx);
+			if (ret == TM_TASK_STATE_STOP || ret == TM_TASK_STATE_UNREGISTERED) {
+				continue;
+			}
+			ret = taskmgr_check_msg_mask(msg, chk_idx);
+			if (ret != OK) {
+				continue;
+			}
+			ret = ioctl(fd, TMIOC_BROADCAST, TASK_PID(chk_idx));
+			if (ret != OK) {
+				continue;
+			}
+			(void)sigqueue(TASK_PID(chk_idx), SIGTM_BROADCAST, msg_broad);
+		}
+	}
+}
+
+static int taskmgr_set_broadcast_handler(int msg_mask, int pid)
+{
+	int ret;
+	ret = taskmgr_get_handle_by_pid(pid);
+	if (ret < 0) {
+		return TM_FAIL_REGISTER;
+	}
+	if (TASK_LIST_ADDR(ret) != NULL) {
+		TASK_MSG_MASK(ret) = msg_mask;
+		return OK;
+	}
+	return TM_FAIL_UNREGISTERED_TASK;
```

Comment:
When taskmgr_get_handle_by_pid returns fail, I think it is unregistered. Is there any chance to fail getting task address even though taskmgr_get_handle_by_pid returns OK?